cpp/src/arrow/compute/exec/asof_join_node.cc

```diff
@@ -1142,13 +1142,15 @@ class AsofJoinNode : public ExecNode {
   static arrow::Result<std::shared_ptr<Schema>> MakeOutputSchema(
       const std::vector<std::shared_ptr<Schema>> input_schema,
       const std::vector<col_index_t>& indices_of_on_key,
-      const std::vector<std::vector<col_index_t>>& indices_of_by_key) {
+      const std::vector<std::vector<col_index_t>>& indices_of_by_key,
+      std::vector<int>* field_output_indices = nullptr) {
```

Comment:
This default `= nullptr` is probably redundant since you supply a value every time you call it.

Reply:
Will fix.

Reply:
Actually, a value is not supplied in [the invocation in line 1300](https://github.com/westonpace/arrow/pull/27/files#diff-5789a42aebc3bd0f5a6db687c78ab0c3eef3e316982f99fe9e0ea21fabed354cR1300), so I'll leave the code as is.

Reply:
Can you add comments for `field_output_indices`? It's not obvious what this variable is supposed to do. (Looks like a output arguments)

Reply:
Will do.